File: 283-move-zeroes/283-move-zeroes.java
class Solution {
    public void moveZeroes(int[] nums) {
        int i=0, j=0;
        while(i!=nums.length && j!=nums.length){
            if (nums[i]==0 && nums[j]!=0){
                swap(nums,i,j);
                i++;
                j++;
            } else if(nums[i]==0 && nums[j]==0){
                j++;
            } else {
                i++;
                j++;
            } 
        }
    }
    
    private void swap(int[] nums, int i, int j){
        int temp=nums[i];
        nums[i]=nums[j];
        nums[j]=temp;
    }
}

File: 283-move-zeroes/283-move-zeroes.go
func moveZeroes(nums []int) {

	for findIndex, zeroIndex := 0, 0; findIndex < len(nums); findIndex++ {
		if v1, v2 := nums[zeroIndex], nums[findIndex]; v1 == 0 && v2 != 0 {
			nums[findIndex], nums[zeroIndex] = nums[zeroIndex], nums[findIndex]
			zeroIndex++
		} else if v1 != 0 {
			zeroIndex++
		}
	}
}